cmd/operator/main.go

```diff
@@ -294,9 +315,29 @@ func run(fs *flag.FlagSet) int {
 		return 1
 	}
 
+	agentScaleSubresourceSupported, err := checkPrerequisites(
```

Comment:
same here, we could use k8sutil.IsAllowed() and move it inside the `if prometheusAgentSupported {...}` block.